### PR TITLE
add metrics

### PR DESCRIPTION
**Pull request by JamesWP**
The docker compose file now starts a metrics gathering backend

The environment variables `ENV_STATSHOST` and `ENV_STATSPORT` allow the stats logging
to be controlled, if they are blank then it's disabled.

The environment variables in the docker compose should be modified before the
interface is made public (e.g. to add a non default password to the grafana frontend)

### CODE REVIEW

**Comment by JamesWP** [comment]
Test by running docker-compose up.
The grafana instance will need a datasource adding, http://graphite/ before the stats are available

File: exchange/exchange_brokerage.h [hunk 1]
```diff
@@ -23,7 +23,8 @@ class Brokerage {
     std::map<std::string, double> accounts_under_management(const std::string& reporting_ccy) const
     {
         std::map<std::string, double> report;
-        for (const auto& [_, account]: accounts) {
+        for (auto& account_entry: accounts) {
```
Comment (nadvamir) [inline exchange/exchange_brokerage.h:26]: I was so sad that in C++ 17 we can't do this without a warning that I didn't remove the warning myself. Thank you.

File: fxbattle/fxbattle_main.cpp [hunk 2]
```diff
@@ -97,16 +96,48 @@ int main(int argc, const char* argv[])
     });
 
     //--------------------------------------------------------------------------
+
+    auto getEnvOrDefault = [](const char *name,
```
Comment (nadvamir) [inline fxbattle/fxbattle_main.cpp:100]: Can you put most of this into a statsd constructor?

File: fxbattle/fxbattle_metrics.h [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      return 1;
+    return statsd_gauge(_link, const_cast<char *>(stat.c_str()), value);
+  }
+
+  int timing(const std::string &stat, std::size_t timeunits) {
+    if (!_link)
+      return 1;
+    return statsd_timing(_link, const_cast<char *>(stat.c_str()), timeunits);
+  }
+/*
```
Comment (nadvamir) [inline fxbattle/fxbattle_metrics.h]: Remove ghost code

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,15 +1,23 @@
 cmake_minimum_required (VERSION 2.6)
+
```
Comment (nadvamir) [inline CMakeLists.txt:2]: Why 3? :)

**Review by nadvamir — CHANGES_REQUESTED** [review]
Great job! Some minor cleanup is still required, but then we're golden :)

File: CMakeLists.txt (hunk 4 above)
Comment (JamesWP) [reply CMakeLists.txt:2]: Moar newlines yo, .. will remove

File: fxbattle/fxbattle_main.cpp (hunk 2 above)
Comment (JamesWP) [reply fxbattle/fxbattle_main.cpp:100]: environment variables are not just for stats, i'll move it into a utils file though

File: fxbattle/fxbattle_metrics.h (hunk 3 above)
Comment (JamesWP) [reply fxbattle/fxbattle_metrics.h]: i think i can just implement it now, there was a reason why i couldn't before but its not here now

File: fxbattle/fxbattle_main.cpp (hunk 2 above)
Comment (nadvamir) [reply fxbattle/fxbattle_main.cpp:100]: getenv - sure, but I also meant the lines 116-123

Comment (JamesWP) [reply fxbattle/fxbattle_main.cpp:100]: i ended up not moving it but moved other noise into stats

**Review by nadvamir — APPROVED** [review]
Let's roll with this :)

**Pull request merged**